bme590_melanoma_detection/main.py

```diff
@@ -151,18 +161,21 @@ def prediction():
     """
     req = request.json
     image = req['image'][23:]
-    imgdata = base64.b64decode(image)
-    try:
-        filename = 'images/some_image' + \
+    if (is_valid_image_string(image))
+        imgdata = base64.b64decode(image) 
+        try:
+            filename = 'images/some_image' + \
                    str(ImPath.query.order_by(ImPath.id.desc()).first().id + 1) + \
                    '.jpg'
-    except:
-        filename = 'images/some_image1.jpg'
-    with open(filename, 'w') as f:
-        f.write(imgdata)
-    image64 = mpl.imread(filename)
-    resp = get_prediction(image64)
-    dictout = {resp[0][0]:str(resp[1][0]), resp[0][1]:str(resp[1][1]), 'impath': filename}
+        except:
+            filename = 'images/some_image1.jpg'
+        with open(filename, 'w') as f:
+            f.write(imgdata)
+        image64 = mpl.imread(filename)
+        resp = get_prediction(image64)
+        dictout = {resp[0][0]:str(resp[1][0]), resp[0][1]:str(resp[1][1]), 'impath': filename}
+    else
+        send_error('Invalid image data', 415)
```

Comment:
I think you have to do `return send_error(...)` for this to work :). Because send error returns a tuple that your handler function here has to return so that flask actually sends the error.